src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -17,15 +17,22 @@ namespace Microsoft.ApplicationInsights.Extensibility.Implementation
     /// <summary>
     /// Runs a task after a certain delay and log any error.
     /// </summary>
-    public class TaskTimer : IDisposable
+    public class TaskTimer
     {
+        private class DelayedWork
+        {
+            public DelayedWork(Func<Task> runFunc) { this.RunFunc = runFunc; this.IsCancellationRequested = false; }
+            public Func<Task> RunFunc { get; private set; }
```

Comment:
"runFunc" is the better name. The Start method parameter ("elapsed") must corresponds to a specific usage of this public class. However, Start being public, I did not want to change the parameter name. Let me call this "runWhenElapsedFunc" to underline the correspondence.